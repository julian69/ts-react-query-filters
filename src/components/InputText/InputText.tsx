import { ChangeEvent, FC } from 'react';

interface InputTextProps {
  value: string;
  label: string;
  onChange: (value: string) => void;
}

const InputText: FC<InputTextProps> = ({ value, label, onChange }) => {
  const id = label.split(' ').join().toLowerCase();
  const handleChange = (event: ChangeEvent<HTMLInputElement>) => onChange(event.target.value);

  return (
    <div className="mr-5">
      <label htmlFor={id} className="text-sm font-medium text-stone-600">{label}</label>
      <input
        id={id}
        type="text"
        value={value}
        onChange={handleChange}
        className="mt-2 block w-full rounded-md border border-gray-100 bg-gray-100 px-2 py-2 shadow-sm outline-none focus:border-blue-500 focus:ring focus:ring-blue-200 focus:ring-opacity-50"
      />
    </div>
  );
};

export default InputText;