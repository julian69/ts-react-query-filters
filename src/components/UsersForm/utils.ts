import { IFilters } from "@/types/users";

export const deserializeUrl = (params: URLSearchParams): IFilters => ({
  name: params.get('name') || '',
  email: params.get('email') || '',
  company: params.get('company') || '',
  zipcode: params.get('zipcode') || '',
});


