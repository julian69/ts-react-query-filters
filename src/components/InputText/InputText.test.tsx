import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@/test/utils';
import { describe, expect, it, vi } from 'vitest';

import InputText from './InputText';

const mockedProps = {
  value: 'test',
  label: 'test',
  onChange: vi.fn(),
};

describe('<InputText />', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<InputText {...mockedProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should show the correct props and trigger the onChange with the right value', () => {
    render(<InputText {...mockedProps} />);
    const expected = 'hi';
    const input = screen.getByDisplayValue(mockedProps.value);
    const value = screen.getByText(mockedProps.value);
    const label = screen.getByText(mockedProps.label);

    expect(value).toBeInTheDocument();
    expect(label).toBeInTheDocument();

    fireEvent.change(input, { target: { value: expected } });

    expect(mockedProps.onChange).toHaveBeenCalledWith(expected);
  });
});