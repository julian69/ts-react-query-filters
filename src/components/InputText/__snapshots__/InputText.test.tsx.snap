// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<InputText /> > should render correctly 1`] = `
<DocumentFragment>
  <div
    class="mr-5"
  >
    <label
      class="text-sm font-medium text-stone-600"
      for="test"
    >
      test
    </label>
    <input
      class="mt-2 block w-full rounded-md border border-gray-100 bg-gray-100 px-2 py-2 shadow-sm outline-none focus:border-blue-500 focus:ring focus:ring-blue-200 focus:ring-opacity-50"
      id="test"
      type="text"
      value="test"
    />
  </div>
</DocumentFragment>
`;
